tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -1838,11 +1826,8 @@ Status CheckInputsWeights(
     const std::vector<std::pair<string, TrtInputArg>>& expected_inputs) {
   const auto& inputs = params.inputs;
   const auto& node_def = params.node_def;
-  if (inputs.size() != expected_inputs.size()) {
-    return errors::InvalidArgument(node_def.op(), " got ", inputs.size(),
-                                   " inputs but expected ",
-                                   expected_inputs.size());
-  }
+  CHECK_INPUT_SIZE(inputs.size(), expected_inputs.size(), node_def);
```

Comment:
This check is not sufficiently general to warrant a macro. Macros should not be preferred over functions in modern C++ code base. It's hiding non-obvious behavior. In Tensorflow, `CHECK_*` macros are used for assertions. This check could be a function and called via 
`TF_RETURN_IF_ERROR(VerifyInputSizes(inputs, expected_inputs, node_def))`

Reply:
You are right, perhaps, the prefix CHECK_ should be replaced, but I think that the use of macros sometimes has some advantages over the C++ functions. For instance, in that case by using macro: 
```
#define CHECK_SHAPE_TENSOR(tensor)                                        \
   if (!IsTrtShapeTensorCompatible(tensor)) {                             \
     TFTRT_ERROR(errors::InvalidArgument, "Tensor of type ",              \
                 DebugString(tensor.dtype()), " having shape ",           \
                 tensor.shape().DebugString(), " is not TRT compatible"); \
   }
```
which uses an other one:
```
#define TFTRT_ERROR(func, ...)                   \
    do {                                         \
      return func("TFTRT::", __FUNCTION__, ":",  \
                    __LINE__, ": ", __VA_ARGS__);\
    } while (0)
```
we have possibility to see the exact place where this check failed, because of  ```__FUNCTION__,  __LINE__```.
We don't have this possibility with the function unless ```__FUNCTION__, __LINE__``` are used as the additional input parameters of 
```
VerifyInputSizes(..., __FUNCTION__, __LINE__)
```
By the way, `TF_RETURN_IF_ERROR` is also a macro, which use an other one: `TF_PREDICT_FALSE`

Reply:
You are right, I didn't realize we are using __LINE__ here. It still seems odd since this macro can only be used in a very particular place. I think it jus shouldn't be moved out of `convert_nodes.cc` as the converters based on the BaseOpConverter shouldn't need this class

Reply:
As you suggested, I renamed macros CHECK_INPUT_SIZE and  CHECK_SHAPE_TENSOR to TFTRT_CHECK_INPUT_SIZE and  TFTRT_CHECK_SHAPE_TENSOR, respectively.
Both these macros use TFTRT_ERROR.
 
As a mater of fact, now TFTRT_CHECK_SHAPE_TENSOR is used in `ops/tile.cc`. It's why now TFTRT_CHECK_SHAPE_TENSOR and TFTRT_ERROR are in `utils.h`

You are right, the other three 
- TFTRT_INTERNAL_ERROR_AT_NODE, 
- TFTRT_RETURN_ERROR_IF_NULLPTR, 
- TFTRT_CHECK_INPUT_SIZE
 
could be moved back to `convert_nodes.cc`. I will do that